{
    "id": "bundle--b1056f0a-5f5c-4ca8-90d7-a2b1fa3c6b66",
    "objects": [
        {
            "created": "2021-08-13T00:21:19.903271Z",
            "created_by_ref": "identity--8ce3f695-d5a4-4dc8-9e93-a65af453a31a",
            "description": "TensorFlow is an end-to-end open source platform for machine learning. In affected versions an attacker can cause undefined behavior via binding a reference to null pointer in all operations of type `tf.raw_ops.MatrixDiagV*`. The [implementation](https://github.com/tensorflow/tensorflow/blob/84d053187cb80d975ef2b9684d4b61981bca0c41/tensorflow/core/kernels/linalg/matrix_diag_op.cc) has incomplete validation that the value of `k` is a valid tensor. We have check that this value is either a scalar or a vector, but there is no check for the number of elements. If this is an empty tensor, then code that accesses the first element of the tensor is wrong. We have patched the issue in GitHub commit f2a673bd34f0d64b8e40a551ac78989d16daad09. The fix will be included in TensorFlow 2.6.0. We will also cherrypick this commit on TensorFlow 2.5.1, TensorFlow 2.4.3, and TensorFlow 2.3.4, as these are also affected and still in supported range.",
            "external_references": [
                {
                    "external_id": "CVE-2021-37657",
                    "source_name": "cve"
                }
            ],
            "id": "vulnerability--f94048da-dc6d-4a54-9985-718aa517c47f",
            "modified": "2021-11-22T14:54:52.900149Z",
            "name": "CVE-2021-37657",
            "spec_version": "2.1",
            "type": "vulnerability"
        }
    ],
    "type": "bundle"
}